project/BuildPlugin.scala

```diff
@@ -222,27 +227,66 @@ object BuildImplementation {
          |  IO.write(file("$testDir/bloop-config/base-directory"), dir.getAbsolutePath)
          |}
          |TaskKey[Unit]("checkInstall") := {
-         |  Thread.sleep(1000) // Let's wait a little bit because of OS's IO latency
-         |  val mostRecentStamp = (bloopConfigDir.value ** "*.config").get.map(_.lastModified).min
-         |  (System.currentTimeMillis() - mostRecentStamp)
-         |  val diff = (System.currentTimeMillis() - mostRecentStamp) / 1000
-         |  if (diff <= 15) () // If it happened in the last 15 seconds, this is ok!
-         |  else sys.error("The sbt plugin didn't write any file")
+         |  val configFiles = (bloopConfigDir.value ** "*.config").get
+         |  if (configFiles.isEmpty) {
+         |    sys.error("The sbt plugin didn't write any file")
+         |  }
          |}
     """.stripMargin
     }
 
+    private val testPluginContents = {
```

Comment:
Is there any reason why we have this here and we don't have it in an independent file in `project`? I think that solution would be more elegant because you would make sure that such code always compiles. You can then copy the file from here to whatever place you need. I'm saying this because it's quite huge, and difficult to parse if it's in a string.

Reply:
It's just that it's much easier to copy to the scripted projects. Having it in `project` means that it would be applied to the projects in our build definition, which I'd rather avoid. Where do you suggest I put it?